src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/AccessRightService.java

```diff
@@ -120,25 +104,63 @@ private AccessRightHolderDto accessRightTypeToDto(AccessRightType accessRightTyp
      * @return
      * @throws ClientNotFoundException if client with given id was not found
      * @throws ServiceNotFoundException if service with given fullServicecode was not found
+     * @throws EndpointNotFoundByServiceNameException if base endpoint for this service is not found from the client
      */
     public List<AccessRightHolderDto> getAccessRightHoldersByService(ClientId clientId, String fullServiceCode)
-            throws ClientNotFoundException, ServiceNotFoundException {
+            throws ClientNotFoundException, ServiceNotFoundException, EndpointNotFoundByServiceNameException {
         ClientType clientType = clientRepository.getClient(clientId);
         if (clientType == null) {
             throw new ClientNotFoundException("Client " + clientId.toShortString() + " not found");
         }
 
         ServiceType serviceType = serviceService.getServiceFromClient(clientType, fullServiceCode);
+        EndpointType endpointType = getEndpoint(clientType, serviceType, EndpointType.ANY_METHOD, EndpointType.ANY_PATH)
+                .orElseThrow(() -> new EndpointNotFoundByServiceNameException(fullServiceCode));
+        return getAccessRightsHoldersByEndpoint(clientType, endpointType);
+    }
 
-        List<AccessRightHolderDto> accessRightHolderDtos = new ArrayList<>();
+    /**
+     * Get access right holders for Endpoint
+     *
+     * @param id
+     * @return
+     * @throws EndpointService.EndpointNotFoundException    if no endpoint is found with given id
+     * @throws ClientNotFoundException                      if client attached to endpoint is not found
+     */
+    public List<AccessRightHolderDto> getAccessRightHoldersByEndpoint(Long id)
+            throws EndpointService.EndpointNotFoundException, ClientNotFoundException {
+        verifyAuthority("VIEW_ENDPOINT_ACL");
```

Comment:
This and other verifyAuthority checks have been removed from AccessRightService

Reply:
There is still 3 verifyAuthorities in `EndpointService` and 1 in `ServiceService`, that seem unnecessary.

There is also a bunch in `ServiceDescriptionService`, but I think they are not unnecessary, since controller layer does not know which permissions it needs. For example ADD_WSDL vs ADD_OPENAPI3 is not determined until we know which type it is. But a better way to do that would be to add for example
`    @PreAuthorize("hasAuthority('ADD_OPENAPI3')")`
to addOpenApi3ServiceDescription instead of verifyAuthority, no real reason why SecurityHelper.verifyAuthority should be used there. Security annotations should be the primary way and verifyAuthority reserved only for exceptional cases.

Reply:
Removed verifyAuthority-calls from EndpointService & ServiceService

Reply:
Thanks! How about that `ServiceDescriptionService` ?

Reply:
Probably my comment was unclear, I though it should replace those `verifyAuthority("EDIT_REST"); `snippets with method level annotations `@PreAuthorize("hasAuthority('EDIT_REST')") `